### ISSUE TITLE
Layer don't receive n don't send messages

### ISSUE THREAD

**Issue by afonsolage** [AragurDEV/yowsup#15]
I'm having a strange issue. Some times my applications stop receiving and sending messages, just after start. Seems like my communication with whatsapp is blocked, but I can authenticate fine.

After sending the message on WhatsApp, I don't receive the ACK (on App on Phone), which means to me the WhatsApp didn't received the ack from my app.

I really don't know where to look to solve this issue.

I'm using YowInterfaceLayer:

    class SendReceiveLayer(YowInterfaceLayer):
        """
        Layer used to Receive and Send message to WhatsApp
        """

        def __init__(self, on_msg_cb):
            """
            :param on_msg_cb: MessageProtocolEntity callback
            """

            print("Starting SR layer.")
            super(SendReceiveLayer, self).__init__()

and the stack:

    class ZapStack(object):
        def __init__(self, credentials, on_msg, encryptionEnabled=True):
            # YowsupEnv.setEnv('android')  # Uses andoid env as default.

            stack_builder = YowStackBuilder()

            self.sr_layer = SendReceiveLayer(on_msg)

            self.stack = stack_builder\
                .pushDefaultLayers(encryptionEnabled)\
                .push(self.sr_layer)\
                .build()

            self.stack.setCredentials(credentials)

        def start(self):
            """
            Start's the stack server, connecting to whatszapp server.
            """
            self.stack.broadcastEvent(YowLayerEvent(
                YowNetworkLayer.EVENT_STATE_CONNECT))
            try:
                self.stack.loop()
            except AuthError as e:
                print("Authentication Error: %s" % e)

**Comment by afonsolage** [comment]
After a while, I receive this:

    Stream Error type: ack
    {'ack': None}

**Comment by AragurDEV** [comment]
On message recieve you have to send back a recieved ack

**Comment by afonsolage** [comment]
I'm sending this, on every message.

    @ProtocolEntityCallback("message")
    def on_message(self, msg_protocol_entity):
        """
        Called when a message is received
        :param msg_protocol_entity: MessageProtocolEntity
        """

        print("Received %s" % msg_protocol_entity)
        self.on_msg_cb(msg_protocol_entity)

        # pass message down to others layers.
        self.toLower(msg_protocol_entity.ack())
        self.toLower(msg_protocol_entity.ack(True))

**Comment by afonsolage** [comment]
I was able to fix it. Indeed I wasn't sending the ACK. But somehow, the cli can retrieve again "unacked" messages.

So when I got this problem during the development, I just run:
    python yowsup-cli demos -e -l "558XXXXXX700:ZLIrAcMXXXXXXXXXXoz38o07Y4k=" -E s40 --yowsup

and later type /L. After it, I receive all pending messages and send back ack. So everything got back to normal.

I'll try to discover what cli does to get the pending acks and I'll add it on startup of my app.

Thanks!
